docs/sanity_functions_reference.rst

```diff
@@ -45,26 +45,35 @@ Currently ReFrame provides three broad categories of sanity functions:
 
    If in the user provided message more placeholders are used than the arguments of the assert function (except the ``msg`` argument), no argument substitution will be performed in the user message.
 3. Utility functions.
-   The are functions that you will normally use when defining :attr:`sanity_patterns <reframe.core.pipeline.RegressionTest.sanity_patterns>` and :attr:`perf_patterns <reframe.core.pipeline.RegressionTest.perf_patterns>`.
    They include, but are not limited to, functions to iterate over regex matches in a file, extracting and converting values from regex matches, computing statistical information on series of data etc.
 
 
-Users can write their own sanity functions as well.
-The page ":doc:`deferrables`" explains in detail how sanity functions work and how users can write their own.
+Users can write their own deferrable functions as well.
+The page ":doc:`deferrables`" explains in detail how deferrable functions work and how users can write their own.
 
 
-.. py:decorator:: sanity_function
+.. py:decorator:: reframe.utility.sanity.deferrable(func)
 
-    Sanity function decorator.
+    Deferrable decorator.
 
-    The evaluation of the decorated function will be deferred and it will become suitable for use in the sanity and performance patterns of a regression test.
+    Converts the decorated free function into a deferrable function.
 
     .. code:: python
 
-        @sanity_function
+        import reframe.utility.sanity as sn
+
+        @sn.deferrable
         def myfunc(*args):
             do_sth()
 
 
+.. py:decorator:: reframe.utility.sanity.sanity_function(func)
+
+    Please use the :func:`reframe.core.pipeline.RegressionMixin.deferrable` decorator when possible. Alternatively, please use the :func:`reframe.utility.sanity.deferrable` decorator instead.
+
+    .. warning:: Not to be mistaken with :func:`~reframe.core.pipeline.RegressionMixin.sanity_function` built-in.
+    .. deprecated:: 3.8.0
```

Comment:
I wanted to play this PR in a conservative way :) There are a few deprecation warnings already for 3.7.0 and I wasn't sure this would go through at any earlier version. We can keep the pin in here to make sure this gets updated properly.